tests/cpu_saturation_test.py

```diff
@@ -0,0 +1,57 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Apache License Version 2.0, see <https://www.apache.org/licenses/>
+# This test is conceived to compare the main branch before and after patching
+# #1175. In the old main:
+# 1. open the process manager / htop, an start monitoring the cpu usage
+# 2. run the test, allow the processes to spawn correctly and take note of the
+#    CPU usage/iterations per second. THERE IS NO NEED TO RUN THE TEST UNTIL THE
+#    END as it does not check anything per se.
+# Run this same test in the patched branch. There should be a significant
+# improvement in performance, due to higher CPU saturation.
+
+from __future__ import annotations
+
+import warnings
+
+import pytest
+import torch
+
+from sbi.inference import simulate_for_sbi
+from sbi.simulators.linear_gaussian import diagonal_linear_gaussian
+from sbi.utils.torchutils import BoxUniform
+from sbi.utils.user_input_checks import process_prior, process_simulator
+
+warnings.simplefilter(action="ignore", category=FutureWarning)
+
+
+def slow_linear_gaussian(theta):
+    """Linear Gaussian simulator with a sleep statement."""
+    x = []
+    for th in theta:
+        # time.sleep(0.05)
+        x.append(diagonal_linear_gaussian(th.reshape(1, -1)))
```

Comment:
is the commented out sleep a left over? 
Also, is the for-loop on purpose to make it slow?

Reply:
The commented out sleep is a left over yes.
The for loop I left in there in case the test wanted to be run with a batched simulation.